pkg/controller/baremetalasset/baremetalasset_controller.go

```
@@ -0,0 +1,113 @@
+var _ reconcile.Reconciler = &ReconcileBareMetalAsset{}
+
+// ReconcileBareMetalAsset reconciles a BareMetalAsset object
+type ReconcileBareMetalAsset struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a BareMetalAsset object and makes changes based on the state read
+// and what is in the BareMetalAsset.Spec
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileBareMetalAsset) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling BareMetalAsset")
+
+	// Fetch the BareMetalAsset instance
+	instance := &appv1alpha1.BareMetalAsset{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Check if the secret exists
+	secretName := instance.Spec.BMC.CredentialsName
+	secret := &corev1.Secret{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: secretName, Namespace: request.Namespace}, secret)
+	if err != nil && errors.IsNotFound(err) {
+		reqLogger.Info("Secret not found", "Namespace", request.Namespace, "Secret.Name", secretName)
+		return reconcile.Result{}, err
```

We don't want to return an error here, because re-trying isn't going to fix anything. This is a perfect state to represent in the `Status`. We can tackle that later, so for now I'd just return `nil`, log the failure as you are, and perhaps add a TODO comment for updating status.

---

Fixed.